node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -635,12 +679,24 @@ async fn process_msg<Context>(
 		CollateOn(id) => {
 			state.collating_on = Some(id);
 		},
-		DistributeCollation(receipt, pov, result_sender) => {
+		DistributeCollation(active_leaf, receipt, pov, result_sender) => {
 			let _span1 = state
 				.span_per_relay_parent
 				.get(&receipt.descriptor.relay_parent)
 				.map(|s| s.child("distributing-collation"));
 			let _span2 = jaeger::Span::new(&pov, "distributing-collation");
+
+			if state.view.leaves().find(|h| **h == active_leaf).is_none() {
```

Comment:
Is it necessarily an active leaf? We can ensure it on collation-generation part, though I don't see a reason to do so.

Reply:
Yes. I think we only have to check whether the relay_parent of the candidate is within constraints (based on our view of active leaves). I don't see a reason why we would want to pass in an active leaf at all.

Reply:
Given that a candidate is no longer bound to be backed on top of a particular parent, all we would need to check is whether there exists at least one active leaf where that candidate can be backed.

Reply:
The active leaf here is not the same as candidate relay parent.
But I believe you're right, we should extend implicit view with a method that allows finding a leaf by hash in its ancestry.

The fragment tree we will try to put the candidate in should be an active leaf, otherwise backing will not accept it.